### ISSUE TITLE
Module is not compatible with upcoming Magento 2.4.6 version

### ISSUE THREAD

**Issue by hostep** [MultiSafepay/magento2#48]
**Describe the bug**
I have access to the Magento 2.4.6-beta2 release and notice that the MultiSafepay module crashes when I try to login to the backoffice.

```
Error: Class "Zend_Http_Client" not found in /Users/pieterhoste/Projects/magento2/m23-demo/vendor/multisafepay/magento2-core/Util/VersionUtil.php:76
Stack trace:
#0 vendor/multisafepay/magento2-core/Util/NotificationUtil.php(73): MultiSafepay\ConnectCore\Util\VersionUtil->getNewVersionsDataIfExist()
#1 vendor/multisafepay/magento2-adminhtml/Observer/BackendAuthUserLoginSuccess.php(58): MultiSafepay\ConnectCore\Util\NotificationUtil->addNewReleaseNotification()
#2 vendor/magento/framework/Event/Invoker/InvokerDefault.php(88): MultiSafepay\ConnectAdminhtml\Observer\BackendAuthUserLoginSuccess->execute(Object(Magento\Framework\Event\Observer))
#3 vendor/magento/framework/Event/Invoker/InvokerDefault.php(74): Magento\Framework\Event\Invoker\InvokerDefault->_callObserverMethod(Object(MultiSafepay\ConnectAdminhtml\Observer\BackendAuthUserLoginSuccess), Object(Magento\Framework\Event\Observer))
#4 vendor/magento/framework/Event/Manager.php(65): Magento\Framework\Event\Invoker\InvokerDefault->dispatch(Array, Object(Magento\Framework\Event\Observer))
#5 generated/code/Magento/Framework/Event/Manager/Proxy.php(95): Magento\Framework\Event\Manager->dispatch('backend_auth_us...', Array)
#6 vendor/magento/module-backend/Model/Auth.php(169): Magento\Framework\Event\Manager\Proxy->dispatch('backend_auth_us...', Array)
#7 vendor/magento/framework/Interception/Interceptor.php(58): Magento\Backend\Model\Auth->login('xxx', 'yyy')
#8 vendor/magento/framework/Interception/Interceptor.php(138): Magento\Backend\Model\Auth\Interceptor->___callParent('login', Array)
#9 vendor/magento/module-admin-adobe-ims/Plugin/DisableAdminLoginAuthPlugin.php(56): Magento\Backend\Model\Auth\Interceptor->Magento\Framework\Interception\{closure}('xxx', 'yyy')
#10 vendor/magento/framework/Interception/Interceptor.php(135): Magento\AdminAdobeIms\Plugin\DisableAdminLoginAuthPlugin->aroundLogin(Object(Magento\Backend\Model\Auth\Interceptor), Object(Closure), 'xxx', 'yyy')
#11 vendor/magento/framework/Interception/Interceptor.php(153): Magento\Backend\Model\Auth\Interceptor->Magento\Framework\Interception\{closure}('xxx', 'yyy')
#12 generated/code/Magento/Backend/Model/Auth/Interceptor.php(59): Magento\Backend\Model\Auth\Interceptor->___callPlugins('login', Array, Array)
#13 vendor/magento/module-backend/App/Action/Plugin/Authentication.php(207): Magento\Backend\Model\Auth\Interceptor->login('xxx', 'yyy')
#14 vendor/magento/module-backend/App/Action/Plugin/Authentication.php(159): Magento\Backend\App\Action\Plugin\Authentication->_performLogin(Object(Magento\Framework\App\Request\Http))
#15 vendor/magento/module-backend/App/Action/Plugin/Authentication.php(127): Magento\Backend\App\Action\Plugin\Authentication->_processNotLoggedInUser(Object(Magento\Framework\App\Request\Http))
#16 vendor/magento/framework/Interception/Interceptor.php(135): Magento\Backend\App\Action\Plugin\Authentication->aroundDispatch(Object(Magento\Backend\Controller\Adminhtml\Index\Index\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#17 vendor/magento/framework/Interception/Interceptor.php(153): Magento\Backend\Controller\Adminhtml\Index\Index\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#18 generated/code/Magento/Backend/Controller/Adminhtml/Index/Index/Interceptor.php(32): Magento\Backend\Controller\Adminhtml\Index\Index\Interceptor->___callPlugins('dispatch', Array, Array)
#19 vendor/magento/framework/App/FrontController.php(245): Magento\Backend\Controller\Adminhtml\Index\Index\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#20 vendor/magento/framework/App/FrontController.php(212): Magento\Framework\App\FrontController->getActionResponse(Object(Magento\Backend\Controller\Adminhtml\Index\Index\Interceptor), Object(Magento\Framework\App\Request\Http))
#21 vendor/magento/framework/App/FrontController.php(147): Magento\Framework\App\FrontController->processRequest(Object(Magento\Framework\App\Request\Http), Object(Magento\Backend\Controller\Adminhtml\Index\Index\Interceptor))
#22 vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\App\FrontController->dispatch(Object(Magento\Framework\App\Request\Http))
#23 vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\App\FrontController\Interceptor->___callParent('dispatch', Array)
#24 vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#25 generated/code/Magento/Framework/App/FrontController/Interceptor.php(23): Magento\Framework\App\FrontController\Interceptor->___callPlugins('dispatch', Array, Array)
#26 vendor/magento/framework/App/Http.php(116): Magento\Framework\App\FrontController\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#27 generated/code/Magento/Framework/App/Http/Interceptor.php(23): Magento\Framework\App\Http->launch()
#28 vendor/magento/framework/App/Bootstrap.php(264): Magento\Framework\App\Http\Interceptor->launch()
#29 pub/index.php(30): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http\Interceptor))
#30 {main}
```

This is due to Magento getting rid of older Zend framework classes in 2.4.6.
See their [Backwards Incompatible page for 2.4.6](https://developer-stage.adobe.com/commerce/php/development/backward-incompatible-changes/highlights/#246-beta) (search for `Zend_HTTP`)

Maybe you guys can already try to figure out a replacement of using Zend_HTTP that's backwards compatible with previous Magento versions.

**To Reproduce**
1. Use Magento 2.4.6-beta2 (can possibly also be reproduced by using the `2.4-develop` branch of magento's github
2. Use MultiSafepay module version 2.22.0
3. Log into the backoffice

**Expected behavior**
Backoffice works & MultiSafepay module as well

**Screenshots**

**Additional context**

**Comment by vinodsowdagar** [comment]
Hi @hostep ,

Thanks for the heads up! I've added this one to the backlog.

**Comment by anvanza** [comment]
Hi @Vinod-MultiSafepay 

How's that backlog item going?
2.4.6 has been released today. Making the module incompatible with the latest Magento version.

**Comment by thlassche** [comment]
I'm also interested in an update for this!

**Comment by vinodsowdagar** [comment]
Hi @hostep , @anvanza , @thlassche ,

Currently there are a couple of changes internally in review and we expect to create a new release at the start of next week.
We will keep you updated once the release has been done.

**Comment by jobkeus** [comment]
@Vinod-MultiSafepay when will the new release be released?

**Comment by Stigmi-Stephane** [comment]
Added a patch on core that make it works : https://github.com/MultiSafepay/magento2-core/pull/23

**Comment by danielcivit** [comment]
Hi @hostep, @anvanza, @thlassche, @jobkeus and @Stigmi-Stephane.

First at all, thanks for contact us. I have news for all of you: 

- Next monday, April 3rd, a new version will be release providing compatibility for Magento 2.4.6.

- @Stigmi-Stephane, thanks for [your contribution](https://github.com/MultiSafepay/magento2-core/pull/23), however the PR you made is going to be rejected. The reason is we are currently providing support for Magento 2.3.X until the latest version released. In order to continue providing backward compatibility use laminas/laminas-http is not an alternative, because this one is not available in Magento 2.3.4 or below. We already found another solution and we are going to remove the dependency with zendframework/zend-http and laminas/laminas-http, to be compatible with any Magento version equal or higher than 2.3.X.
  
- New release is also going to be a major release, since we refactored the notification webhook process. Notification webhooks are now processed in the core module and not in the frontend module and now are faster and cleaner. 

==

You might want to consider as a temporary solution the PR provided by @Stigmi-Stephane, or maybe you want to wait a bit longer now that you know our plan in detail. Our apologies for not provide an answer before.

===

This ticket will remain open until the release date, and we will get back to you to confirm is done. In the meantime, feel free to share any feedback or question you might still have.

**Comment by vinodsowdagar** [comment]
Hi @hostep , @anvanza , @thlassche , @jobkeus , @Stigmi-Stephane ,

We have released a new version of the module which adds support for Magento version 2.4.6: https://github.com/MultiSafepay/magento2/releases/tag/3.0.0

Please note that this is a major version introducing backwards incompatibility.
We strongly advise to first test this version on a staging environment first, before deploying this on production.

To install this version of the module, use the following composer command:
`composer require multisafepay/magento2:3.0.0  --with-all-dependencies`

For now i will close this issue.
If there are more questions or comments, please create a new issue.
